controllers/operatorpolicy_controller.go

```diff
@@ -1446,8 +1448,11 @@ func (r *OperatorPolicyReconciler) musthaveInstallPlan(
 	}
 
 	if len(approvableInstallPlans) != 1 {
-		changed := updateStatus(policy,
-			installPlanUpgradeCond(allUpgradeVersions, approvableInstallPlans), relatedInstallPlans...)
+		changed := updateStatus(
+			policy,
+			installPlanUpgradeCond(complianceConfig, allUpgradeVersions, approvableInstallPlans),
```

Comment:
@JustinKuli this seems like an odd case where OperatorPolicy can't perform an upgrade but the user wouldn't be notified based on `complianceConfig`. I think this particular case should always be NonCompliant. What do you think?